app/react-client-app/src/components/EditTable.jsx

```diff
@@ -1,130 +1,175 @@
-                </Flex>
-            </div>
-        )
+    if (date_event) {
+      setDate(date_event.name);
     }
+    setTable(table);
+  }, []);
+
+  const updateTable = () => {
+    setLoadingMessage("Updating table");
+    // update the table then redirect
+    const person_id = fields.find((field) => field.name == selectedPerson);
+    const date_event = fields.find((field) => field.name == selectedDate);
+    const data = {
+      person_id: person_id.id,
+      date_event: date_event.id,
+    };
+    usePatch(`/scanreporttables/${value}/`, data)
+      .then((res) => {
+        // redirect
+        window.location.href = `/scanreports/${table.scan_report}/`;
+      })
+      .catch((err) => {
+        setAlert({
+          hidden: false,
+          status: "error",
+          title: "Could not update scan report table",
+          description: err.statusText ? err.statusText : "",
+        });
+        onOpen();
+      });
+  };
+  if (!table || !fields || loadingMessage) {
+    //Render Loading State
     return (
-        <div>
-            <CCBreadcrumbBar>
-                <Link href={"/"}>Home</Link>
-                <Link href={"/scanreports/"}>Scan Reports</Link>
-                <Link href={`/scanreports/${scanReport.current.id}/`}>{scanReport.current.dataset}</Link>
-                <Link href={`/scanreports/${scanReport.current.id}/tables/${table.id}/`}>{table.name}</Link>
-                <Link href={`/scanreports/${scanReport.current.id}/tables/${table.id}/update/`}>Update</Link>
-            </CCBreadcrumbBar>
-            <PageHeading text={"Update Table"} />
-            {isOpen &&
-                <ScaleFade initialScale={0.9} in={isOpen}>
-                    <ToastAlert hide={onClose} title={alert.title} status={alert.status} description={alert.description} />
-                </ScaleFade>
-            }
-            <CCSelectInput
-                id={"table-person-id"}
-                label={"Person ID"}
-                selectOptions={fields.map((item) => item.name)}
-                handleInput={setPerson}
-                isDisabled={!canEdit}
-            />
-            <CCSelectInput
-                id={"table-date-event"}
-                label={"Date Event"}
-                selectOptions={fields.map((item) => item.name)}
-                handleInput={setDate}
-                isDisabled={!canEdit}
-            />
-            {canEdit &&
-                <Button
-                    bgColor="#3db28c"
-                    mt="10px"
-                    onClick={updateTable}
-                >
-                    Update {table.name}
-                </Button>
-            }
-        </div>
+      <div>
+        <Flex padding="30px">
+          <Spinner />
+          <Flex marginLeft="10px">
+            {loadingMessage ? loadingMessage : "Loading page"}
+          </Flex>
+        </Flex>
+      </div>
     );
-}
+  }
+  return (
+    <div>
+      <CCBreadcrumbBar>
```

Comment:
One for the future, but it just reminds me we should at some point scrub the `CC` prefix (which was `coconnect`) to something more appropriate.